compiler/src/iree/compiler/Codegen/Common/GPU/GPUHeuristics.cpp

```diff
@@ -279,8 +295,8 @@ static GPUMMASchedule getOptimalMMASchedule(const GPUMatmulShapeType &problem,
   // 16x16x16 intrinsic, then:
   //  - mTotalTileCounts would be 4 * (16/16) = 4
   //  - nTotalTileCounts would be 2 * (32/16) = 4
-  SmallVector<int64_t> mTotalTileCounts = problem.mSizes;
-  SmallVector<int64_t> nTotalTileCounts = problem.nSizes;
+  SmallVector<int64_t, 2> mTotalTileCounts = problem.mSizes;
+  SmallVector<int64_t, 2> nTotalTileCounts = problem.nSizes;
```

Comment:
Why do you add size as type versus the original `SmallVector<int64_t>`?

@hanhanW Told me to not use those in IREE in a previous code review [here](https://github.com/iree-org/iree/pull/19739#discussion_r1925470078), and quote:

> In IREE, we don't specify N for SmallVector. See https://discourse.llvm.org/t/psa-smallvector-t-just-works/57153

Reply:
I think this case is an exception as it didnt just work, becuase of the struct becoming too large and I got the error posted in the PR description.

Reply:
I always see 
```
  SmallVector<GPUMatmulShapeType, 2> intrinsics;
  intrinsics.reserve(target.getWgp().getMma().size());
  ```
  I think both of these lines are fine as theyh are reserving something explicit, the problem the compiler had with just 
  ```
    SmallVector<GPUMatmulShapeType> intrinsics;
   ```
   is that this implictly reserves N=6, i.e 6 of these struct elements which becomes rather large so it wants us to mention that explicitly, if we say  for e.g `SmallVector<GPUMatmulShapeType, 100> intrinsics;` I think the compiler would still be okay with it as we are explicitly asking for that much allocation.

Reply:
Okay thanks for clarifying. We should probably address compiler's complaints more directly with either:
 - SmallVector<GPUMatmulShapeType, 256> intrinsics;
 - Or, SmallVector<GPUMatmulShapeType> intrinsics; intrinsices.reserve(\<some heuristic that can decide the size better\>)

The point being, allocating a small vector of size 2 is no different than the default (N=6). In a way, it is worse because if we always need 256 as size vector size, re-allocating every time when it reaches small vector size limit is going to be expensive.